sql/information_schema/columns_table.go

```diff
@@ -52,29 +52,39 @@ var typeToNumericPrecision = map[query.Type]int{
 // ColumnsTable describes the information_schema.columns table. It implements both sql.Node and sql.Table
 // as way to handle resolving column defaults.
 type ColumnsTable struct {
-	name    string
-	schema  sql.Schema
-	catalog sql.Catalog
+	TableName   string
+	TableSchema sql.Schema
+	catalog     sql.Catalog
 	// allColsWithDefaultValue is the full schema of all tables in all databases. We need this during analysis in order
 	// to resolve the default values of some columns, so we pre-compute it.
 	allColsWithDefaultValue sql.Schema
 
-	rowIter func(*sql.Context, sql.Catalog, sql.Schema) (sql.RowIter, error)
+	RowIter func(*sql.Context, sql.Catalog, sql.Schema) (sql.RowIter, error)
 }
 
 var _ sql.Table = (*ColumnsTable)(nil)
 var _ sql.StatisticsTable = (*ColumnsTable)(nil)
 var _ sql.Databaseable = (*ColumnsTable)(nil)
 var _ sql.DynamicColumnsTable = (*ColumnsTable)(nil)
 
+// HandleColumnsTable is used by Doltgres to inject its correct schema and row
+// iter. In Dolt, this just returns the current columns table implementation.
+var HandleColumnsTable = func() *ColumnsTable {
```

Comment:
Call this NewColumnsTable

Reply:
Probably also want to declare the func separately with its own name `newMySQLColumnsTable`, then declare the var to reference that func.